### PR TITLE
#159987735 highlight and comment on article

### PR DESCRIPTION
**Pull request by gonuoha**
add article highlight-comment feature

### Description of Task to be completed?
add tests for highlighted-comment
add highlighted and cssId columns to comments model
modify create comments controller to update article on highlighted-comment
modify get comments model to include highlighted and cssId

### How should this be manually tested?
Run 'npm install'
create a .env using the .env.sample template
Run 'npm run migrate'
Run 'npm run start:dev'
use postman to test:
post: /api/articles/article_slug/comments
get: /api/articles/article_slug/comments/

### What are the relevant pivotal tracker stories?
[#159987735](https://www.pivotaltracker.com/story/show/159987735)

### CODE REVIEW

File: server/controllers/comments.js [hunk 1]
```diff
@@ -22,7 +111,13 @@ class CommentsController {
   static createComment(req, res, next) {
     const slug = req.params.article_slug;
     const { userId } = req;
-    const { comment } = req.body;
+    const {
```
Comment (emmaadesile) [inline server/controllers/comments.js:114]: Good work but you can have lines 112 and 113 on one line using object destructuring. 
`const { userId, body: { comment } } = req;`

File: server/tests/controllers/comments.js [hunk 2]
```diff
@@ -118,6 +148,82 @@ describe('Comment Controller', () => {
...
+          done();
+        });
+    });
+
+    it('should return error if cssId is missing', (done) => {
+      request(app)
+        .post('/api/articles/test-article-slug12345/comments')
+        .set('Authorization', token)
+        .send(highlightedComment.noCssId)
+        .end((err, res) => {
```
Comment (emmaadesile) [inline server/tests/controllers/comments.js:188]: Object destructuring makes your code more concise. You can apply this on lines `189-193`
const { type, status, body } = res;
You can even go a step further and do everything in one destructuring pattern
`const { type, status, body: {status}, body: { error: { cssId } } } = res;` 
But if it becomes less readable and hairy, simple destructuring would do.

File: server/tests/controllers/comments.js [hunk 3]
```diff
@@ -118,6 +148,82 @@ describe('Comment Controller', () => {
...
+          done();
+        });
+    });
+
+    it('should return error if injected body mismatch', (done) => {
+      request(app)
+        .post('/api/articles/test-article-slug12345/comments')
+        .set('Authorization', token)
+        .send(highlightedComment.mismatch)
+        .end((err, res) => {
```
Comment (emmaadesile) [inline server/tests/controllers/comments.js:203]: Destructuring can also be applied on lines `204-208`

File: server/tests/controllers/comments.js [hunk 4]
```diff
@@ -118,6 +148,82 @@ describe('Comment Controller', () => {
...
+          done();
+        });
+    });
+
+    it('should create a highlighted-comment to an article', (done) => {
+      request(app)
+        .post('/api/articles/test-article-slug12345/comments')
+        .set('Authorization', token)
+        .send(highlightedComment.complete)
+        .end((err, res) => {
```
Comment (emmaadesile) [inline server/tests/controllers/comments.js:218]: You can also use destructuring from lines `220-222`

**Review by emmaadesile — CHANGES_REQUESTED** [review]

File: server/tests/controllers/comments.js (hunk 2 above)
Comment (AdmiralOnu) [reply server/tests/controllers/comments.js:188]: Thank you Emmanuel. We used the dot accessor `res.type` and `res.body.status` for consistency within the comments tests file. However, I do think object destructuring is more cryptic in this case. Do you think we should proceed with this change?

**Review by CEOehis — APPROVED** [review]
LGTM

**Pull request merged**